contracts/VotingEscrow.vy

```diff
@@ -597,6 +593,28 @@ def force_withdraw():
     log Supply(supply_before, supply_before - value)
 
 
+@external
+@nonreentrant('lock')
+def migrate():
+    assert(self.next_ve_contract != ZERO_ADDRESS) # dev: no next ve contract
+
+    _locked: LockedBalance = self.locked[msg.sender]
+    assert block.timestamp < _locked.end, "lock expired"
+    value: uint256 = convert(_locked.amount, uint256)
+
+    ERC20(self.token).approve(self.next_ve_contract, value)
+    Migrator(self.next_ve_contract).migrateLock(msg.sender, value)
+
+    old_locked: LockedBalance = _locked
+    _locked.end = 0
+    _locked.amount = 0
+    self.locked[msg.sender] = _locked
+    supply_before: uint256 = self.supply
+    self.supply = supply_before - value
+    self._checkpoint(msg.sender, old_locked, _locked)
```

Comment:
Question:

Should we clear the `next_ve_contract` value after migrating?

Reply:
Here the users are the one doing the migration it's not possible to clear the value.